### DISCUSSION TITLE
Deleting all admin users causes lockout from Keycloak instance

### DISCUSSION THREAD

**Discussion by jmlaubach** [keycloak/keycloak#20424] [Q&A]
Currently, any and all admins in Keycloak can be deleted by any admin user. This obviously causes a total lockout from logging into Keycloak. I feel like this shouldn't be possible, but I have not found a solution to recover/create a new admin while testing this scenario. Research shows the add-user-keycloak script can add a new admin user, however this is deprecated and no longer part of Keycloak, particularly the Docker image which I'm running. The CLI commands first require authentication to do anything it seems, which is not possible if all admins are deleted.

Maybe there is something I'm missing but I have not found any mention of a way to create new admins if they are deleted or to prevent users from being deleted. I'm running the newest 21.1.1 version. If anyone knows an easy way to do this, would be nice to know the best way to address this scenario should it come up.

**Comment by vmuzikar** [comment]
This should be resolved as part of https://github.com/keycloak/keycloak/issues/19913

**Reply by jmlaubach** [reply]
Great thank you. Just to clarify, I didn't notice a specific solution mentioned in that thread from what I saw. Is there a solution implemented or a workaround that can be used right now until something is implemented? I see that the admin will automatically be recreated as long as no other users exist in that master realm, but I don't want to rely on that workaround.

**Reply by vmuzikar** [reply]
There is no workaround yet, I'm afraid. We want to handle this properly and not rush something in, since admin recovery is a security sensitive area.

**Reply by jmlaubach** [reply]
Sounds good, thanks.
